src/pages/Progress/Progress.jsx

```diff
@@ -0,0 +1,58 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getUserProgressData } from '../../data/firebase';
+import ProgressGrid from '../../components/ProgressGrid/ProgressGrid';
+import BackButton from '../../components/BackButton/BackButton';
+import classes from './Progress.module.css';
+
+class Progress extends Component {
+  constructor({ location }) {
+    super({ location });
+    this.location = location;
```

Comment:
you don't need to do this

Reply:
I fixed this. Now I destruct ```location``` in render.
```suggestion
class Progress extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: null,
    };
    this.getData();
  }
```